fuse/ipns/ipns_unix.go

```diff
@@ -309,279 +271,130 @@ func (s *Node) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
-
-	r, err := uio.NewDagReader(ctx, s.Nd, s.Ipfs.DAG)
-	if err != nil {
-		return err
-	}
-	o, err := r.Seek(req.Offset, os.SEEK_SET)
-	lm["res_offset"] = o
+	fisize, err := fi.fi.Size()
 	if err != nil {
 		return err
 	}
 
-	buf := resp.Data[:min(req.Size, int(r.Size()))]
-	n, err := io.ReadFull(r, buf)
+	readsize := min(req.Size, int(fisize-req.Offset))
+	n, err := io.ReadFull(fi.fi, resp.Data[:readsize])
 	resp.Data = resp.Data[:n]
-	lm["res_size"] = n
 	return err // may be non-nil / not succeeded
 }
 
-func (n *Node) Write(ctx context.Context, req *fuse.WriteRequest, resp *fuse.WriteResponse) error {
-	// log.Debugf("ipns: Node Write [%s]: flags = %s, offset = %d, size = %d", n.name, req.Flags.String(), req.Offset, len(req.Data))
-	if IpnsReadonly {
-		log.Debug("Attempted to write on readonly ipns filesystem.")
-		return fuse.EPERM
-	}
-
-	if n.dagMod == nil {
-		// Create a DagModifier to allow us to change the existing dag node
-		dmod, err := mod.NewDagModifier(ctx, n.Nd, n.Ipfs.DAG, n.Ipfs.Pinning.GetManual(), chunk.DefaultSplitter)
-		if err != nil {
-			return err
-		}
-		n.dagMod = dmod
-	}
-	wrote, err := n.dagMod.WriteAt(req.Data, int64(req.Offset))
+func (fi *File) Write(ctx context.Context, req *fuse.WriteRequest, resp *fuse.WriteResponse) error {
+	wrote, err := fi.fi.WriteAt(req.Data, req.Offset)
 	if err != nil {
 		return err
 	}
 	resp.Size = wrote
-	return nil
-}
 
-func (n *Node) Flush(ctx context.Context, req *fuse.FlushRequest) error {
-	if IpnsReadonly {
-		return nil
-	}
-
-	// If a write has happened
-	if n.dagMod != nil {
-		newNode, err := n.dagMod.GetNode()
-		if err != nil {
-			return err
-		}
-
-		if n.parent != nil {
-			log.Error("updating self in parent!")
-			err := n.parent.update(n.name, newNode)
-			if err != nil {
-				log.Criticalf("error in updating ipns dag tree: %s", err)
-				// return fuse.ETHISISPRETTYBAD
-				return err
-			}
-		}
-		n.Nd = newNode
-
-		/*/TEMP
-		dr, err := mdag.NewDagReader(n.Nd, n.Ipfs.DAG)
-		if err != nil {
-			log.Critical("Verification read failed.")
-		}
-		b, err := ioutil.ReadAll(dr)
-		if err != nil {
-			log.Critical("Verification read failed.")
-		}
-		fmt.Println("VERIFICATION READ")
-		fmt.Printf("READ %d BYTES\n", len(b))
-		fmt.Println(string(b))
-		fmt.Println(b)
-		//*/
-
-		n.dagMod = nil
-
-		n.wasChanged()
-	}
 	return nil
 }
 
-// Signal that a node in this tree was changed so the root can republish
-func (n *Node) wasChanged() {
-	if IpnsReadonly {
-		return
-	}
-	root := n.nsRoot
-	if root == nil {
-		root = n
-	}
-
-	root.repub.Publish <- struct{}{}
+func (fi *File) Flush(ctx context.Context, req *fuse.FlushRequest) error {
+	return fi.fi.Close()
 }
 
-func (n *Node) republishRoot() error {
-
-	// We should already be the root, this is just a sanity check
-	var root *Node
-	if n.nsRoot != nil {
-		root = n.nsRoot
-	} else {
-		root = n
-	}
+func (fi *File) Fsync(ctx context.Context, req *fuse.FsyncRequest) error {
+	return fi.fi.Flush()
```

Comment:
ok it's weird that `Flush(...)` calls `fi.Close()`, but `Fsync(...)` calls `fi.Flush()`. pls explain it in a comment